lnwire/dyn_ack.go

```diff
@@ -144,3 +147,43 @@ func (da *DynAck) Decode(r io.Reader, _ uint32) error {
 func (da *DynAck) MsgType() MessageType {
 	return MsgDynAck
 }
+
+// Validate does DynAck signature validation for a given prior DynPropose
+// message.
+func (da *DynAck) Validate(propose DynPropose, nextHeight uint64,
+	pubkey *secp256k1.PublicKey) error {
+
+	cSig, err := da.Sig.ToSignature()
+	if err != nil {
+		return err
+	}
+
+	var msg bytes.Buffer
+	err = WriteChannelID(&msg, da.ChanID)
+	if err != nil {
+		return err
+	}
+
+	err = WriteElement(&msg, nextHeight)
+	if err != nil {
+		return err
+	}
+
+	tlvData, err := propose.SerializeTlvData()
+	if err != nil {
+		return err
+	}
+
+	msg.Write(tlvData)
+
+	digest := chainhash.DoubleHashB(msg.Bytes())
+
+	if !cSig.Verify(digest, pubkey) {
```

Comment:
FWIW, sig validation isn't present in this package yet. We usually have some higher level package figure that out (gossip messages, commit sig, etc).